### PR TITLE
Leaderboard

### PR DESCRIPTION
**Pull request by dhale27**
Created leaderboard functionality. Leaderboard is displayed but is not being saved after each run. Next sprint will incorporate pickle for file creation.

### CODE REVIEW

File: source/CardGames.py [hunk 1]
```diff
@@ -176,12 +167,34 @@ def calculateHand(self):
...
+    result_card = None
+    for card in self.hand:
+      if(card.value > highest):
+        highest = card.value
+        result_card = card
+    return result_card
+
+  def bust(self):
+    return self.calculateHand() > 21
+
```
Comment (roygarrett) [inline source/CardGames.py:224]: This is all duplicates. Needs to be removed before merging.

File: source/CardGames.py [hunk 2]
```diff
@@ -1,9 +1,8 @@
-
 import random
 import os
 
 cardImages = []
-values = [11, 2, 3, 4, 5, 6, 7, 8, 9, 10, 10, 10, 10]
+values = list(range(1,14))
 suits = ["Spades", "Clubs", "Hearts", "Diamonds"]
```
Comment (roygarrett) [inline source/CardGames.py]: This is old. It needs to stay how it is on the left for blackjack values. Fix this before merging

File: source/CardGames.py [hunk 3]
```diff
@@ -203,3 +216,23 @@ def dealCards(self, numCards: int, players: "list[Player]"):
...
+        player.display()
+        if player.bust() == True:
+          print("You bust!\n")
+          players.pop(players.index(player))
+      else:
+        print("%s passed.\n" % (player.name))
+        players_passed.append(player)
+        players.pop(players.index(player))
+      input("Next Player press 'Enter' when ready!")
+  return players_passed
```
Comment (roygarrett) [inline source/CardGames.py]: This is already in the dev_0 branch remove this before merging

File: test/LEADERBOARD_FILE
Comment (roygarrett) [inline test/LEADERBOARD_FILE:1]: Maybe remove this since we will not be using it this sprint

File: test/test_leaderboard.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    print ('Winner: %s' %(winner.name))
+    
+    
+    updateLeaderboard(winner)
+
+    showLeaderboard(wins)
+ 
+test_leaderboard()
+test_leaderboard()
+
```
Comment (roygarrett) [inline test/test_leaderboard.py:35]: Not sure why you call the test function twice. Also there is a lot of unnecessary spacing. Just stylistic though doesn't affect functionality.

**Review by roygarrett — COMMENTED** [review]
It seems like you might have something weird going on with your dev_0 branch since there's a bunch of stuff it says you are adding that is old or already there. We can figure this out next Tuesday.

**Review by bennettapps — APPROVED** [review]

File: source/CardGames.py (hunk 3 above)
Comment (dhale27) [reply source/CardGames.py]: This is my feature brah

**Review by roygarrett — APPROVED** [review]

**Pull request merged**